packages/compiler/src/core/helpers/type-name-utils.ts

```diff
@@ -182,7 +182,16 @@ function getInterfaceName(iface: Interface, options: TypeNameOptions | undefined
 }
 
 function getOperationName(op: Operation, options: TypeNameOptions | undefined) {
-  return `${getNamespacePrefix(op.namespace, options)}${getIdentifierName(op.name, options)}`;
+  let opName = getIdentifierName(op.name, options);
+  if (op.node.templateParameters.length > 0) {
+    // template
+    const params = op.node.templateParameters.map((t) => getIdentifierName(t.id.sv, options));
```

Comment:
@timotheeguerin , is there any case that template instantiated is true for operation's template arguments? I can only find it's true when the parent interface is used by alias, so skipped that part here for operation.

And about the template instantiated, is it only true when it's used by alias? I found for some statement like "op Op1 is Op2", the template instantiated is still false for Op2. is that expected?

Reply:
not exactly sure what you mean by the template is instantiated but this are how things should be:
- `op Action<T>(t: T): void;` // this `Action` is a template declaration
- `alias A = Action<string>`; ` // `Action<string>` is now a template instance and then so is `A` as its just an alias of it
- `op my is Action<string>`; // `Action<string>` is still a template instance but `my` is not it is just an operation without template params and shoudln't expose them

Reply:
by "template is instantiated", I mean the `isTemplateInstance(iface)` function would return true. But I found in your 3rd case, `isTemplateInstance(iface)` function would still return false for `Action<string>`, is that expected?

Reply:
yeah `isTemplateInstance(Action<String>)` should return true so if it doesn't its a bug
`isTemplateInstance(my) === false` though is expected